factory/processing/processComponents.go

```diff
@@ -1011,20 +1084,109 @@ func (pcf *processComponentsFactory) indexGenesisBlocks(genesisBlocks map[uint32
 		pcf.statusComponents.OutportHandler().SaveBlock(arg)
 	}
 
+	genesisMiniBlockHeaderHandlers, err := pcf.createGenesisMiniBlockHandlers(miniBlocks)
+	if err != nil {
+		return err
+	}
+
 	log.Info("indexGenesisBlocks(): historyRepo.RecordBlock", "shardID", currentShardId, "hash", genesisBlockHash)
-	// TODO: save also genesis body transactions into node storage
-	err = pcf.historyRepo.RecordBlock(genesisBlockHash, genesisBlockHeader, &dataBlock.Body{}, nil, nil, nil, nil)
+	if txsPoolPerShard[currentShardId] != nil {
+		err = pcf.historyRepo.RecordBlock(
+			genesisBlockHash,
+			originalGenesisBlockHeader,
+			genesisBody,
+			txsPoolPerShard[currentShardId].Scrs,
+			txsPoolPerShard[currentShardId].Receipts,
+			intraShardMiniBlocks,
+			txsPoolPerShard[currentShardId].Logs)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = pcf.saveGenesisMiniBlocksToStorage(miniBlocks)
 	if err != nil {
 		return err
 	}
 
+	if txsPoolPerShard[currentShardId] != nil {
+		err = pcf.saveGenesisTxsToStorage(txsPoolPerShard[currentShardId].Txs)
+		if err != nil {
+			return err
+		}
+	}
+
 	nonceByHashDataUnit := dataRetriever.GetHdrNonceHashDataUnit(currentShardId)
 	nonceAsBytes := pcf.coreData.Uint64ByteSliceConverter().ToByteSlice(genesisBlockHeader.GetNonce())
 	err = pcf.data.StorageService().Put(nonceByHashDataUnit, nonceAsBytes, genesisBlockHash)
 	if err != nil {
 		return err
 	}
 
+	// save altered genesis block into storage
```

Comment:
can you extract the altered genesis block header construction and save into a different method (L1126-L2253)? 
and call that here

Reply:
done